### PR TITLE
more numbers

### LINKED ISSUE #155: Inconclusive type checker error message
A simple code to reproduce the error:

```
add(a: Nat, b: Nat): Nat
  case a
  { 
    zero: b
    succ: add(a.pred,Nat.succ(b))
  }

abc:_ 
add(3,4)
//comment
```

Get an error message of Expected letter in the code below it, but it doesn't explain the error.

The correct code cannot have a comment below the last line:

```
abc:_ 
add(3,4)
```

### PR DESCRIPTION
**Pull request by rigille**
this PR implements functions and literals for many more machine integers, including two's complement signed integers, F32 and F64 (as primitives). it also updates literals in the whole codebase, and solves issue https://github.com/uwu-tech/Kind/issues/155#issue-851934910.
should we keep backward compatibility with the old literals? right now they're removed.
the compiler optimizations are pending.

### CODE REVIEW

File: base/Word/shl.kind [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Word.shl<size: Nat>(n: Word(size), value: Word(size)): Word(size)
+  let n_nat = Word.to_nat!(n)
+  Word.shift_right!(n_nat, value)
```
Comment (dumblob) [inline base/Word/shl.kind]: Please don't take me wrong. Could you elaborate why `.shift_right` in the context of `shl`?

Comment (rigille) [reply base/Word/shl.kind]: no reason why, that's just a bug 🙂
good catch, it should be fixed now

**Comment by rigille** [comment]
old literals won't be removed but will be deprecated

**Comment by VictorTaelin** [comment]
Tasty. When it will be merged?

**Pull request merged**